src/Tools/Test Framework/Test Runner/src/DataDrivenTest/DataInputs/TestInputGroup.Table.al

```diff
@@ -87,6 +154,70 @@ table 130454 "Test Input Group"
         Rec.Insert(true);
     end;
 
+    local procedure UpdateIndentation()
+    begin
+        if Rec."Parent Group Code" = '' then
+            Rec.Indentation := 0
+        else
+            Rec.Indentation := 1;
+    end;
+
+    local procedure UpdateGroup()
+    var
+        ExistingGroup: Record "Test Input Group";
+        DefaultGroup: Record "Test Input Group";
+        AllGroupsWithSameName: Record "Test Input Group";
+    begin
+        // Only update parent groups if this record doesn't already have a parent
+        if Rec."Parent Group Code" <> '' then
+            exit;
+
+        // Find if there's already a group with the same Group Name
+        ExistingGroup.SetRange("Group Name", Rec."Group Name");
+        ExistingGroup.SetFilter(Code, '<>%1', Rec.Code);
+
+        if not ExistingGroup.FindFirst() then
+            exit; // No existing group with same name, nothing to do
+
+        // Check if current record is default language - it should become parent of all
+        if Rec."Language ID" = GetDefaultLanguageID() then begin
+            // This is default language, make it the parent of all others with same Group Name
+            AllGroupsWithSameName.SetRange("Group Name", Rec."Group Name");
+            AllGroupsWithSameName.SetFilter(Code, '<>%1', Rec.Code);
+            if AllGroupsWithSameName.FindSet(true) then
+                repeat
+                    AllGroupsWithSameName."Parent Group Code" := Rec.Code;
+                    AllGroupsWithSameName.Modify(true);
+                until AllGroupsWithSameName.Next() = 0;
+        end else begin
+            // This is not default language, find the default language group and make it the parent
+            DefaultGroup.SetRange("Group Name", Rec."Group Name");
+            DefaultGroup.SetRange("Language ID", GetDefaultLanguageID());
+            if DefaultGroup.FindFirst() then
+                Validate(Rec."Parent Group Code", DefaultGroup.Code)
+            else begin
+                // No default language version exists yet, use the first existing group that has no parent as parent
+                ExistingGroup.SetRange("Group Name", Rec."Group Name");
+                ExistingGroup.SetFilter(Code, '<>%1', Rec.Code);
+                ExistingGroup.SetRange("Parent Group Code", '');
+                if ExistingGroup.FindFirst() then
+                    Validate(Rec."Parent Group Code", ExistingGroup.Code);
+            end;
+        end;
+    end;
+
+    procedure GetTestInputGroupLanguages(var LanguageVersions: Record "Test Input Group"): Boolean
+    begin
+        LanguageVersions.Reset();
+        LanguageVersions.SetRange("Group Name", Rec."Group Name");
+        exit(LanguageVersions.FindSet());
+    end;
+
+    local procedure GetDefaultLanguageID(): Integer
+    begin
+        exit(1033); // en-US
```

Comment:
Shouldn't it be blank? I think we should distinguish between en-US and blank. If we want to use integers it can be zero, but blank would be better.

Reply:
Having a default could be changed globally. If you want to enforce the language, it is OK. 
I was just wondering do we have any scenarios with blank (not specified value). If not we can drop it.